src/Cli/dotnet/OptionForwardingExtensions.cs

```diff
@@ -19,7 +19,11 @@ public static class OptionForwardingExtensions
         public static ForwardedOption<T> ForwardAsSingle<T>(this ForwardedOption<T> option, Func<T, string> format) => option.SetForwardingFunction(format);
 
         public static ForwardedOption<string[]> ForwardAsProperty(this ForwardedOption<string[]> option) => option
-            .SetForwardingFunction((optionVals) => optionVals.SelectMany(optionVal => new string[] { $"{option.Aliases.FirstOrDefault()}:{optionVal.Replace("roperty:", string.Empty)}" }));
+            .SetForwardingFunction((optionVals) =>
+                optionVals
+                    .Select(optionVal => optionVal.Replace(";", "%3B")) // must escape semicolon-delimited property values when forwarding them to MSBuild
```

Comment:
this is critical - without escaping (which we do in several other places in the codebase we probably should try to unify), msbuild will eat the `;` character. This is a longstanding issue that was helpfully pointed out by @janv8000 [here](https://github.com/dotnet/sdk/issues/26026#issuecomment-1161431464) that I boorishly disregarded. many thanks, friend!

as a result now, we consistently handle all msbuild-known property option forms.